server/src/main/java/de/uftos/services/TeacherService.java

```diff
@@ -44,7 +44,30 @@ public TeacherService(TeacherRepository repository) {
   public Page<Teacher> get(Pageable pageable, Optional<String> firstName,
                            Optional<String> lastName, Optional<String> acronym,
                            Optional<String[]> subjects, Optional<String[]> tags) {
-    return this.repository.findAll(pageable);
+    Specification<Teacher> spec = Specification.where(null);
+    firstName.ifPresent((param) -> spec.or(createFilter(param, "firstName")));
+    lastName.ifPresent((param) -> spec.or(createFilter(param, "lastName")));
+    acronym.ifPresent((param) -> spec.or(createFilter(param, "acronym")));
+    subjects.ifPresent((param) -> {
+      for (String subjectId : param) {
+        spec.and(dropDownFilter(subjectId, "subjects"));
+      }
+    });
+    tags.ifPresent((param) -> {
+      for (String tagId : param) {
+        spec.and(dropDownFilter(tagId, "tags"));
```

Comment:
You should write spec = spec.and... because spec.and returns a new Specification instance. Specification is just like String, immutable. This should be changed everywhere where you have spec.or or spec.and (And therefore you can not use a function needing a lamba like ifPresent